src/main/java/org/prebid/server/cache/CoreCacheService.java

```diff
@@ -553,4 +585,44 @@ private BidCacheRequest toBidCacheRequest(List<CachedCreative> cachedCreatives)
                 .map(CachedCreative::getPayload)
                 .toList());
     }
+
+    private String resolveCacheKey(String accountId) {
+        return resolveCacheKey(accountId, null);
+    }
+
+    // hbCacheId is accepted here to have a backwards-compatibility with video category mapping
+    private String resolveCacheKey(String accountId, String hbCacheId) {
+        if (!appendTraceInfoToCacheId) {
+            return hbCacheId;
+        }
+
+        // no need to have additional separator if datacenter name won't be added
+        final boolean isDatacenterNamePopulated = StringUtils.isNotBlank(datacenterRegion);
+        final int separatorCount = isDatacenterNamePopulated ? 2 : 1;
+        final int accountIdLength = accountId.length();
+        final int traceInfoLength = isDatacenterNamePopulated
+                ? accountIdLength + datacenterRegion.length() + separatorCount
+                : accountIdLength + separatorCount;
+
+        final String cacheKey = hbCacheId != null ? hbCacheId : idGenerator.generateId();
+        if (cacheKey != null && traceInfoLength < cacheKey.length()) {
+            final String substring = cacheKey.substring(0, cacheKey.length() - traceInfoLength);
+            return isDatacenterNamePopulated
+                    ? accountId + TRACE_INFO_SEPARATOR + datacenterRegion + TRACE_INFO_SEPARATOR + substring
+                    : accountId + TRACE_INFO_SEPARATOR + substring;
+        } else {
+            return hbCacheId;
+        }
+    }
+
+    private String normalizeDatacenterRegion(String datacenterRegion) {
+        if (datacenterRegion == null) {
+            return null;
+        }
+
+        final String trimmedDatacenterRegion = datacenterRegion.trim();
+        return trimmedDatacenterRegion.length() > MAX_DATACENTER_REGION_LENGTH
+                ? trimmedDatacenterRegion.substring(0, MAX_DATACENTER_REGION_LENGTH)
+                : trimmedDatacenterRegion;
```

Comment:
Lets move this logic to spring configuration, so that we don't do same computation every request.

Reply:
This method is called only once during bean creation. Or you want to avoid the situation of someone misusing the method?

Reply:
my bad, nevermind.